starknet_py/net/gateway_client.py

```diff
@@ -210,15 +210,24 @@ async def estimate_fee(
 
     async def call_contract(
         self,
-        invoke_tx: Union[InvokeFunction, Call],
+        call: Call = None,  # pyright: ignore
         block_hash: Optional[Union[Hash, Tag]] = None,
         block_number: Optional[Union[int, Tag]] = None,
+        *,
+        invoke_tx: Call = None,  # pyright: ignore
     ) -> List[int]:
-        if isinstance(invoke_tx, InvokeFunction):
+        if call is not None and invoke_tx is not None:
+            raise ValueError("invoke_tx and call are mutually exclusive")
+
+        if invoke_tx is not None:
             warnings.warn(
-                "InvokeFunctions has been deprecated as a call_contract parameter, use Call instead.",
+                "invoke_tx parameter is deprecated. Use call instead",
                 category=DeprecationWarning,
             )
+            call = call or invoke_tx
+
+        if call is None:
+            raise ValueError("Call is a required parameter")
```

Comment:
This logic is repeated multiple times, creating more surface for bugs and inconsistencies. I suggest adding a helper private method to Client that would accept `call` and `invoke_tx` and would return call to be used.